docs/concepts/overview/extending.md

```diff
@@ -0,0 +1,210 @@
+Plugins](https://github.com/kubernetes/community/blob/master/contributors/devel/flexvolume.md)
+and [Network
+Plugins](https://kubernetes.io/docs/concepts/cluster-administration/network-plugins/))
+and by kubectl.  Plugin interfaces still evolving, and may change.
+
+Below is a diagram showing how the extensions points interact with the
+Kubernetes control plane.
+
+<img src="https://docs.google.com/drawings/d/e/2PACX-1vQBRWyXLVUlQPlp7BvxvV9S1mxyXSM6rAc_cbLANvKlu6kCCf-kGTporTMIeG5GZtUdxXz1xowN7RmL/pub?w=960&h=720">
+
+<!-- image source drawing https://docs.google.com/drawings/d/1muJ7Oxuj_7Gtv7HV9-2zJbOnkQJnjxq-v1ym_kZfB-4/edit?ts=5a01e054 -->
+
+
+## Extension Points
+
+This diagram shows the extension points in a Kubernetes system.  
+
+<img src="https://docs.google.com/drawings/d/e/2PACX-1vSH5ZWUO2jH9f34YHenhnCd14baEb4vT-pzfxeFC7NzdNqRDgdz4DDAVqArtH4onOGqh0bhwMX0zGBb/pub?w=425&h=809">
+
+<!-- image source diagrams: https://docs.google.com/presentation/d/1lho38WMS3spo0euY_l_jsrknSeOqJkeGheMohI7R7F0/edit#slide=id.g1ffd17cda3_0_123 -->
+
+*   Kubectl plugins extend the kubectl binary ①.  [Link for 1.8 docs branch: docs/tasks/extend-kubectl/kubectl-plugins ] They only affect the individual user's local environment.  These extension points are described in the [Client Extensions](todo) section.
+*   The apiserver ②  handles all requests.  Several types of extension points in the apiserver allow authenticating requests, or blocking them based on their content, editing content, and handling deletion.  These are described in the [API Access Extensions](todo) section. 
+*   The apiserver serves various kinds of **resources**, ③.  **Built-in resource kinds**, like `pods`, are defined by the Kubernetes project and have to be .     You can also add resources that you definde are several ways to add **Custom Resources** to your API-Server.  Custom Resources are often used with API Access Extensions.
+*   The kubernetes scheduler, ④, decides which nodes to place pods on.  There are several ways to extend scheduling.  These are described in the [Scheduler Extensions](bloop) section.
+*   Much of the behavior of Kubernetes is implemented by client programs Controllers, ⑤.  You can write your own, as described in the [Controllers](boo) section.  Controllers are often used in conjunction with User-defined Types.
+*   The kubelet runs on servers, and mediates access to storage, ⑥, and the network, ⑦.  These are described in the [Infrastructure Extensions](poof) section.
+
+If you are unsure where to start, this flowchart can help.  Note that some solutions may involve several types of extensions.
+
+
+
+<img source="https://docs.google.com/drawings/d/e/2PACX-1vRWXNNIVWFDqzDY0CsKZJY3AR8sDeFDXItdc5awYxVH8s0OLherMlEPVUpxPIB1CSUu7GPk7B2fEnzM/pub?w=1440&h=1080">
+
+<!-- image source drawing: https://docs.google.com/drawings/d/1sdviU6lDz4BpnzJNHfNpQrqI9F19QZ07KnhnxVrp2yg/edit --> 
+
+## API Extensions
+### User-Defined Types
```

Comment:
is this CRD? if so might be good to use the same terminology.

Reply:
Yes, Fixed.